online-store/src/components/view/viewApp.test.ts

```diff
@@ -0,0 +1,339 @@
+    quantity: 2,
+    year: 2019,
+    manufacturer: 'Samsung',
+    color: 'белый',
+    cams: 3,
+    favorite: false,
+  },
+  {
+    id: 4,
+    name: 'Xiaomi Poco X4 pro',
+    quantity: 1,
+    year: 2022,
+    manufacturer: 'Xiaomi',
+    color: 'желтый',
+    cams: 3,
+    favorite: false,
+  },
+  {
+    id: 5,
+    name: 'Xiaomi Redmi Note 11',
+    quantity: 4,
+    year: 2022,
+    manufacturer: 'Xiaomi',
+    color: 'белый',
+    cams: 3,
+    favorite: true,
+  },
+];
+
+const dataSortedByYearAsc: IItem[] = [
+  {
+    id: 2,
+    name: 'Samsung Galaxy A03',
+    quantity: 5,
+    year: 2000,
+    manufacturer: 'Samsung',
+    color: 'красный',
+    cams: 2,
+    favorite: false,
+  },
+  {
+    id: 7,
+    name: 'Apple iPhone 11',
+    quantity: 12,
+    year: 2018,
+    manufacturer: 'Apple',
+    color: 'белый',
+    cams: 3,
+    favorite: true,
+  },
+  {
+    id: 1,
+    name: 'Samsung Galaxy S21',
+    quantity: 2,
+    year: 2019,
+    manufacturer: 'Samsung',
+    color: 'белый',
+    cams: 3,
+    favorite: false,
+  },
+  {
+    id: 8,
+    name: 'Apple iPhone 13',
+    quantity: 10,
+    year: 2020,
+    manufacturer: 'Apple',
+    color: 'красный',
+    cams: 2,
+    favorite: false,
+  },
+  {
+    id: 5,
+    name: 'Xiaomi Redmi Note 11',
+    quantity: 4,
+    year: 2022,
+    manufacturer: 'Xiaomi',
+    color: 'белый',
+    cams: 3,
+    favorite: true,
+  },
+  {
+    id: 4,
+    name: 'Xiaomi Poco X4 pro',
+    quantity: 1,
+    year: 2022,
+    manufacturer: 'Xiaomi',
+    color: 'желтый',
+    cams: 3,
+    favorite: false,
+  },
+];
+
+const dataSortedByQtyAsc: IItem[] = [
+  {
+    id: 4,
+    name: 'Xiaomi Poco X4 pro',
+    quantity: 1,
+    year: 2022,
+    manufacturer: 'Xiaomi',
+    color: 'желтый',
+    cams: 3,
+    favorite: false,
+  },
+  {
+    id: 1,
+    name: 'Samsung Galaxy S21',
+    quantity: 2,
+    year: 2019,
+    manufacturer: 'Samsung',
+    color: 'белый',
+    cams: 3,
+    favorite: false,
+  },
+  {
+    id: 5,
+    name: 'Xiaomi Redmi Note 11',
+    quantity: 4,
+    year: 2022,
+    manufacturer: 'Xiaomi',
+    color: 'белый',
+    cams: 3,
+    favorite: true,
+  },
+  {
+    id: 2,
+    name: 'Samsung Galaxy A03',
+    quantity: 5,
+    year: 2000,
+    manufacturer: 'Samsung',
+    color: 'красный',
+    cams: 2,
+    favorite: false,
+  },
+  {
+    id: 8,
+    name: 'Apple iPhone 13',
+    quantity: 10,
+    year: 2020,
+    manufacturer: 'Apple',
+    color: 'красный',
+    cams: 2,
+    favorite: false,
+  },
+  {
+    id: 7,
+    name: 'Apple iPhone 11',
+    quantity: 12,
+    year: 2018,
+    manufacturer: 'Apple',
+    color: 'белый',
+    cams: 3,
+    favorite: true,
+  },
+];
+
+describe('AppView.renderItem', () => {
+  it('Should be an instance of Function', () => {
+    expect(AppView.renderItem).toBeInstanceOf(Function);
+  });
+
+  it('Should return rendered unselected item', () => {
+    //
+    expect(AppView.renderItem(data[0])).toMatchSnapshot();
```

Comment:
In real production I would stick with philosophy of https://testing-library.com/docs/
Example from my head: you better render component, then click on the button and try to test result of your click. (It might be filter results). So these tests are more bullet-proof